### ISSUE TITLE
Error invoking method, Failed to launch JVM. (UPDATE: Program crashes if game is installed on non-default disk)

### ISSUE THREAD

**Issue by Reilley64** [SeanPesce/Dark_Souls_Config_Utility#9]
I was getting these errors when trying to launch the program, so I ran it through a CMD and it outputs this.

>Checking for Dark Souls install...
Checking for custom Steam Library directory...
Checking Registry for Dark Souls install...
Checking game version...
Exception in Application start method
java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at com.sun.javafx.application.LauncherImpl.launchApplicationWithArgs(LauncherImpl.java:389)
        at com.sun.javafx.application.LauncherImpl.launchApplication(LauncherImpl.java:328)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at sun.launcher.LauncherHelper$FXHelper.main(Unknown Source)
Caused by: java.lang.RuntimeException: Exception in Application start method
        at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:917)
        at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$154(LauncherImpl.java:182)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NullPointerException
        at dscfgutil.view.DSCfgMainUI.checkDSVersion(DSCfgMainUI.java:1428)
        at dscfgutil.view.DSCfgMainUI.checkForDS(DSCfgMainUI.java:1369)
        at dscfgutil.view.DSCfgMainUI.initWindow(DSCfgMainUI.java:430)
        at dscfgutil.view.DSCfgMainUI.initUI(DSCfgMainUI.java:384)
        at dscfgutil.view.DSCfgMainUI.<init>(DSCfgMainUI.java:376)
        at dscfgutil.DSCfgUtil.start(DSCfgUtil.java:30)
        at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$161(LauncherImpl.java:863)
        at com.sun.javafx.application.PlatformImpl.lambda$runAndWait$174(PlatformImpl.java:326)
        at com.sun.javafx.application.PlatformImpl.lambda$null$172(PlatformImpl.java:295)
        at java.security.AccessController.doPrivileged(Native Method)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$173(PlatformImpl.java:294)
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at com.sun.glass.ui.win.WinApplication.lambda$null$147(WinApplication.java:177)
        ... 1 more
Exception running application dscfgutil.DSCfgUtil

**Comment by SeanPesce** [comment]
**EDIT:** Issue fixed in [v0.944](https://github.com/SeanPesce/Dark_Souls_Config_Utility/releases)  


------------------------------------  

**Original response:**  

Thanks for the heads up! This was a very early project for me so it has a ton of old, poorly-written code, and the program often crashes when encountering unexpected configurations for DSFix, PvP Watchdog, etc. I'd recommend trying the following:  

 * **Make sure you're using the [latest version](https://github.com/SeanPesce/Dark_Souls_Config_Utility/releases)** of the program, as the last release fixed a ton of crashes.  

 * **Install/run the program as an administrator**, because some features may require admin privileges depending on your version of Windows and your UAC level.  

 * If the other options don't work, your best bet is to try removing any DSFix/PvP Watchdog files from your DATA folder, as a previous configuration could be causing issues. Most importantly, **make sure the configuration files (DSfix.ini, DSfixKeys.ini, and DSPWSteam.ini) are deleted** from the DATA folder before launching.  

Unfortunately I won't be making any more major updates to the program, because the code is a pain to work with and the Remastered edition of the game will be making the project obsolete anyway.  That being said, please let me know whether you get it working! Again, I really appreciate the bug report.

**Comment by Reilley64** [comment]
I've tried all the options that you suggested. Even reinstalling Dark Souls so that I only had the base game didn't work. This happened when moving my game install from my HDD to my SSD, wondering if that could've tripped something. Thanks for the help.

**Comment by SeanPesce** [comment]
> This happened when moving my game install from my HDD to my SSD, wondering if that could've tripped something.  

Ah yep, that's probably it. This might work:  

 * Go to your Steam configuration file (mine is **C:\Program Files (x86)\Steam\config\config.vdf**) and open it in a text editor (Notepad, Notepad++, etc).  

* Find the entries that start with "**BaseInstallFolder**". If there's more than 1, make sure the first one is the folder with Dark Souls in it. For example, if you have:  

```
...
"BaseInstallFolder_1"    "D:\\Games"
"BaseInstallFolder_2"    "E:\\Games"
...
```
 * Then Dark Souls should be in **D:\\Games\\Dark Souls Prepare to Die Edition**.  

If that's the issue, it would only take a few lines of code to fix, so if you have the time to check it out that'd be super helpful.

**Comment by Reilley64** [comment]
I only seem to have one base install folder on my HDD, which seems weird as steam is installed on my SSD and I do have other games installed on it. When I move Dark Souls back to the HDD the program does run as normal.

**Comment by SeanPesce** [comment]
# [Fixed in v0.944](https://github.com/SeanPesce/Dark_Souls_Config_Utility/releases)
